packages/core/src/workflows/default.ts

```diff
@@ -159,23 +162,58 @@ export class DefaultExecutionEngine extends ExecutionEngine {
       delay?: number;
     };
     runtimeContext: RuntimeContext;
+    parentAISpan?: AnyAISpan;
     abortController: AbortController;
     writableStream?: WritableStream<ChunkType>;
   }): Promise<TOutput> {
-    const { workflowId, runId, graph, input, resume, retryConfig } = params;
+    const { workflowId, runId, graph, input, resume, retryConfig, runtimeContext, parentAISpan } = params;
     const { attempts = 0, delay = 0 } = retryConfig ?? {};
     const steps = graph.steps;
 
     //clear runCounts
     this.runCounts.clear();
 
+    const spanArgs = {
+      name: `workflow run: '${workflowId}'`,
+      input,
+      attributes: {
+        workflowId,
+      },
+    };
+
+    // if parentSpan passed, use it to build workflowSpan
+    // otherwise, attempt to create new trace
+    let aiSpan: AISpan<AISpanType.WORKFLOW_RUN> | undefined;
+    if (parentAISpan) {
+      aiSpan = parentAISpan.createChildSpan({
+        type: AISpanType.WORKFLOW_RUN,
+        ...spanArgs,
+      });
+    } else {
+      const aiTracing = getSelectedAITracing({
+        runtimeContext: runtimeContext,
+      });
+      if (aiTracing) {
+        aiSpan = aiTracing.startSpan({
+          type: AISpanType.WORKFLOW_RUN,
+          ...spanArgs,
+          startOptions: {
+            runtimeContext,
+          },
+        });
+      }
+    }
```

Comment:
entry point for workflows.  if `aiTracing` is undefined, then workflows won't be traced.